src/warp.jl

```diff
@@ -87,28 +71,117 @@ julia> indices(imgr)
 (Base.OneTo(906),Base.OneTo(905))
 ```
 """
-function warp(img::AbstractExtrapolation, tform)
-    inds = autorange(img, tform)
-    out = OffsetArray(Array{eltype(img)}(map(length, inds)), inds)
+function warp_new{T}(img::AbstractExtrapolation{T}, tform, inds::Tuple = autorange(img, inv(tform)))
```

Comment:
Presumably invertibility is more of a trait?  I can't see it fitting neatly into the type hierarchy.